### ISSUE TITLE
I can't debug build with Flutter 2.0.0

### ISSUE THREAD

**Issue by y1row** [adjust/flutter_sdk#42]
When I try to build in the environment of Flutter 2.0.0, this error message appears and it seems that I can not build. Is there any workaround?

```
Ld /Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Products/Debug-iphoneos/Adjust/Adjust.framework/Adjust normal (in target 'Adjust' from project 'Pods')
    cd /Users/user/.ghq/github.com/user1/app/ios/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-ios11.0 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.4.sdk -L/Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Products/Debug-iphoneos/Adjust -F/Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Products/Debug-iphoneos/Adjust -F/Users/user/flutter/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-armv7_arm64 -filelist /Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/Adjust.build/Objects-normal/arm64/Adjust.LinkFileList -install_name @rpath/Adjust.framework/Adjust -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/Adjust.build/Objects-normal/arm64/Adjust_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -fembed-bitcode -Xlinker -bitcode_verify -Xlinker -bitcode_hide_symbols -Xlinker -bitcode_symbol_map -Xlinker /Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Products/Debug-iphoneos/Adjust -fobjc-arc -fobjc-link-runtime -framework SystemConfiguration -weak_framework AdSupport -weak_framework CoreTelephony -weak_framework iAd -framework Flutter -framework Foundation -framework SystemConfiguration -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/Adjust.build/Objects-normal/arm64/Adjust_dependency_info.dat -o /Users/user/Library/Developer/Xcode/DerivedData/Runner-fmijjoaoeppsqsacgnydsiaqeana/Build/Products/Debug-iphoneos/Adjust/Adjust.framework/Adjust
    
ld: bitcode bundle could not be generated because '/Users/user/flutter/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-armv7_arm64/Flutter.framework/Flutter' was built without full bitcode. All frameworks and dylibs for bitcode must be generated from Xcode Archive or Install build file '/Users/user/flutter/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-armv7_arm64/Flutter.framework/Flutter'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

pubspec.yaml
```
adjust_sdk: ^4.26.0
```

Flutter
```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.0, on Mac OS X 10.15.7 19H512 darwin-x64, locale ja-JP)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.41.1)
[✓] Connected device (3 available)
```

**Comment by yelkamel** [comment]
Same here

**Comment by uerceg** [comment]
Hey guys.

I am not sure right now if this has anything to do with Adjust plugin. I have created new app with Flutter 2 and noticed this error when I try to run iOS app from Xcode (or Android Studio on iOS device). Sounds like it might be related to [this issue](https://github.com/flutter/flutter/issues/48092). Applying what is suggested in [this comment](https://github.com/flutter/flutter/issues/48092#issuecomment-577345215) solves the error for me. What I did in particular for running from Xcode is that I went to edit scheme of my target (Runner) and set **Build Configuration** to **Release**:

<img width="1904" alt="Screenshot 2021-03-19 at 01 53 51" src="https://user-images.githubusercontent.com/1741635/111716299-01946700-8856-11eb-9052-f81576e81cfd.png">

Looks like it can be workaround, but I will try to investigate this a bit in more detail since according to [this comment](https://github.com/flutter/flutter/issues/48092#issuecomment-777009170) seems like some kind of fix for this should be out there, but I still managed to see this error with latest stable Flutter version.

Let us know if this worked for you.

Cheers

**Comment by y1row** [comment]
Hi @uerceg .

Thank you for information.
When I changed the scheme to Release in Xcode, I was able to build in Xcode. 🎉 

But it doesn't seem to be able to build from Android Studio. The same message is displayed.
I can't develop it unless the debug build is started, so I hope it works.

**Comment by uerceg** [comment]
I will try to investigate this further. In the meantime and for the record, opened up the [issue](https://github.com/flutter/flutter/issues/78589) in Flutter repo as well on this topic. Will keep you posted.

**Comment by uerceg** [comment]
At the end, seems like it's an issue with Flutter engine and that they are working on the fix for the issue. In the meantime, you can give [suggested workaround](https://github.com/flutter/flutter/issues/78589#issuecomment-803009380) a shot.

**Comment by yelkamel** [comment]
thanks @uerceg  for you help.

Off course debug release have to work to develop and test feature properly.
I hope Flutter team will solve it asap.

**Comment by Fatimamostafa** [comment]
Thanks for raising the concern. Took me hours to figure out the issue. And now going with the workaround. Working fine.

**Comment by bgoktugozdemir** [comment]
![Screen Shot 2021-04-05 at 13 25 48](https://user-images.githubusercontent.com/26827403/113564684-73203380-9612-11eb-97cc-2db7114d9d5a.png)

I updated the Adjust_Sdk package to 4.28.0, but it still has this problem.

**Comment by uerceg** [comment]
Hi @bgoktugozdemir 

That's expected because bug is not in Adjust SDK, but rather Flutter platform itself. When they release a new Flutter version which will have fix for this, that's when the things will start to work.

Cheers

**Comment by bgoktugozdemir** [comment]
> Hi @bgoktugozdemir
> 
> That's expected because bug is not in Adjust SDK, but rather Flutter platform itself. When they release a new Flutter version which will have fix for this, that's when the things will start to work.
> 
> Cheers

Thank you for your answer. 

What can I for iOS until the new version? Cannot I use the Adjust package?

Thanks

**Comment by yelkamel** [comment]
Hello @uerceg just to let you know there is a warning on a deprecated thing:  


`/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/adjust_sdk-4.28.0/android/src/main/java/com/adjust/sdk/flutter/AdjustSdk.java:49:` warning: [deprecation] Registrar in PluginRegistry has been deprecated
import io.flutter.plugin.common.PluginRegistry.Registrar;
                                              
/Users/yelkamel/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/adjust_sdk-4.28.0/android/src/main/java/com/adjust/sdk/flutter/AdjustSdk.java:120: warning: [deprecation] Registrar in PluginRegistry has been deprecated
    public static void registerWith(Registrar registrar) {`

**Comment by uerceg** [comment]
Hi @bgoktugozdemir 

Have you tried the [workaround](https://github.com/flutter/flutter/issues/78589#issuecomment-803009380)? Also seems like Flutter released version 2.0.4 where this issue might be addressed.

Hi @yelkamel 

Yes. We are aware. This is from the time when Flutter changed infrastructure plugin was relying on and then we pretty much left logic for older Flutter versions in place for users who might be using those older ones + added new API for users who use latest versions. Flutter community is usually pretty up to date with Flutter releases so that old code might not be relevant anymore. We'll take a look and see if we can remove it. But other than inconvenience of seeing this deprecation warning message, there should not be any impact to functionality or anything like that.

Cheers

**Comment by uerceg** [comment]
Back to original topic of this issue - we have just tested latest Flutter platform update (2.0.4) and we were able to successfully build debug version of the app. Feel free to give it a shot, it should work. Thank you all for input and patience in here. Also thanks to @jmagman for assistance on the issue and 2.0.4 Flutter update. Cheers guys.
